test/functional/home_controller_test.rb

```diff
@@ -141,13 +141,4 @@ def setup
     get :dashboard_v2
     assert_includes response.body, alert
   end
-
-  test 'trending tags are returned when a user has not subscribed to any topics' do
-    current_user = users(:user_without_tag_subscriptions)
-    expected_tag = Tag.trending.pluck(:name).sample
-    UserSession.create(current_user)
-    get :dashboard_v2
-    assert_includes response.body, expected_tag
-  end
-
```

Comment:
The reason this test keeps [failing](https://github.com/publiclab/plots2/pull/9133/checks?check_run_id=1839068000) is because of how I first hard code the expected tags on line 147 and pick out one of the tag names "randomly"

This query is then run separately in the home controller https://github.com/publiclab/plots2/blob/2188e0a4d9430e0c8f9f60db00cb54bd74d6ef94/app/controllers/home_controller.rb#L163-L166

Therefore, if line 147 in the test file returns a result, there is no guarantee that it will be the same as when line 165 in the home controller file is run, that is why the expected, is not the same as the returned value, therefore the test fails(https://github.com/publiclab/plots2/pull/9133/checks?check_run_id=1839068000) 

The trending tags will always be different for all users depending on the subscriptions they have on the site. 

In the meantime, I am trying to identify what part of the trending query would return different results each time... 
```
  def self.trending(limit = 5, start_date = DateTime.now - 1.month, end_date = DateTime.now)
    Tag.select('term_data.tid, term_data.name, term_data.count') # ONLY_FULL_GROUP_BY, issue #8152 & #3120
       .joins(:node_tag, :node)
       .where('node.status = ?', 1)
       .where('node.created > ?', start_date.to_i)
       .where('node.created <= ?', end_date.to_i)
       .distinct
       .order('count DESC')
       .limit(limit)
  end
```

Reply:
This test is better off as a count > 0 and not suited for specificity since the tag names returned are not consistent.